kore/src/Kore/Equation/Equation.hs

```diff
@@ -262,3 +273,32 @@ substitute assignments equation =
         }
   where
     Equation { requires, left, right, ensures, attributes } = equation
+
+{- | The list of identifiers for an 'Equation'.
+
+The identifiers are:
+
+* the rule label
+* symbol names on the left-hand side
+* symbol 'Klabel's on the left-hand side
```

Comment:
why do we need symbol KLabels? Aren't symbol names enough?

Reply:
Reading below, they seem to be useful for identifying the rule when rewriting. Is that the case?

Reply:
A typical user won't know the symbol names, just the K labels. Allowing both ways of specifying a rule is an affordance to users and developers both.